observation_portal/requestgroups/duration_utils.py

```diff
@@ -194,10 +197,10 @@ def get_complete_configurations_duration(configurations_list, start_time, priori
                 if configuration_dict['guiding_config']['mode'] in request_overheads['guiding_overheads']:
                     duration += request_overheads['guiding_overheads'][configuration_dict['guiding_config']['mode']]
 
-            # TODO: find out if we need to have a configuration type change time for spectrographs?
-            if configdb.is_spectrograph(configuration_dict['instrument_type']):
-                if previous_conf_type != configuration_dict['type']:
-                    duration += request_overheads['config_change_overhead']
+            # Certain Configuration Types for certain Instrument Types will have a non-zero config_change_overhead.
+            # For instance, this could account for Lamp startup times when first switching to an ARC or LAMP_FLAT configuration.
+            if previous_conf_type != configuration_dict['type']:
+                duration += configuration_types[configuration_dict['type']]['config_change_overhead']
```

Comment:
Removed the check for if it is a spectrograph. Instead any configuration_type that is switched to will have its config_change_overhead added, but for most types that will be zero. If you only have a single configuration_type and it has a non-zero config_change_overhead, then it will not be applied since you aren't changing types, but I think that can get covered by the front_padding. If you think not, we could change this logic to apply the config_change_overhead for the "first" configuration too...

Reply:
Wait, doesn't the `config_change_overhead` always apply to the first configuration with this logic? On the first pass through the loop, `previous_conf_type` is an empty string so won't match any configuration. I do think it makes sense to apply the change overhead to the first configuration, it seems like that would allow us to model the overheads better. But were you thinking it would just be better to capture it with the front padding?

Reply:
you are correct, I had assumed we pre-populated with the first configuration type but we don't. It is probably better this way, since it should more accurately model times...